AoE2ScenarioParser/objects/support/new_effect.py

```diff
@@ -1565,16 +1566,25 @@ def modify_attribute_by_variable(
             object_attributes: int | None = None,
             variable: int | None = None,
             message: str | None = None,
+            armour_attack_class: int | None = None,
     ):
-        return self._trigger_ref._add_effect(
+        if (armour_attack_class is not None and object_attributes != ObjectAttribute.ATTACK and
+                object_attributes != ObjectAttribute.ARMOR):
+            raise ValueError("Cannot use 'armour_attack_class' for non attack/armor attributes.")
+        effect = self._trigger_ref._add_effect(
             EffectId.MODIFY_ATTRIBUTE_BY_VARIABLE,
             object_list_unit_id=object_list_unit_id,
             source_player=source_player,
             operation=operation,
             object_attributes=object_attributes,
             variable=variable,
-            message=message
+            message=message,
         )
+        if armour_attack_class is not None and armour_attack_class > -1:
+            combined_var = effect._aa_to_quantity(variable, armour_attack_class)
```

Comment:
might need to refac tor so this method is more easily accessible here

Reply:
also it is more intuitive that the least significant bits would be the 2nd argument of the function. currently its the reverse